### ISSUE TITLE
SFT Loss?

### ISSUE THREAD

**Issue by hkunzhe** [NVlabs/VILA#29]
Thanks for the great work! I fine-tuned the VILA 8B model using my own instruction-tuning data (mostly videos) according to [the script](https://github.com/NVlabs/VILA/blob/main/scripts/v1_5/release/8b/3_sft.sh) with the default params. I noticed that the initial training loss fluctuated between 1.5-1.6, and then the loss suddenly increased to 6-7. Can you tell me what the training loss convergence curve observed in your SFT experiments?

Any advices are appreciated!

**Comment by jpWang** [comment]
+1
I have also tried to fine-tune vila-1.5-llama3-8b with video instruction tuning data for the video caption task, but there is no obvious sign of a decrease in loss.
Looking forward to your reply~
@yaolug

**Comment by jpWang** [comment]
> +1 I have also tried to fine-tune vila-1.5-llama3-8b with video instruction tuning data for the video caption task, but there is no obvious sign of a decrease in loss. Looking forward to your reply~ @yaolug

the command I used:
```
torchrun --nnodes 1 --nproc_per_node 8 --master_port 25001 --master_addr $MASTER_ADDR --node_rank $NODE_RANK llava/train/train_mem.py \
    --deepspeed ./scripts/zero3.json \
    --model_name_or_path ../Llama-3-VILA1.5-8B \
    --version llama_3 \
    --data_mixture $DATA_NAME \
    --vision_tower ../siglip-so400m-patch14-384 \
    --mm_vision_select_feature cls_patch \
    --mm_projector mlp_downsample \
    --tune_vision_tower True \
    --tune_mm_projector True \
    --tune_language_model True \
    --mm_vision_select_layer -2 \
    --mm_use_im_start_end False \
    --mm_use_im_patch_token False \
    --image_aspect_ratio resize \
    --bf16 True \
    --output_dir ./checkpoints/$OUTPUT \
    --num_train_epochs 1 \
    --per_device_train_batch_size 8 \
    --gradient_accumulation_steps 1 \
    --evaluation_strategy "no" \
    --save_strategy "steps" \
    --save_steps 500 \
    --save_total_limit 10 \
    --learning_rate 2e-5 \
    --weight_decay 0. \
    --warmup_ratio 0.03 \
    --lr_scheduler_type "cosine" \
    --logging_steps 1 \
    --tf32 True \
    --model_max_length 2560 \
    --gradient_checkpointing True \
    --dataloader_num_workers 8 \
    --lazy_preprocess True \
    --vflan_no_system_prompt True \
    --report_to tensorboard  \
```

**Comment by yaolug** [comment]
The increase of loss to 6-7 is definitely not reasonable. In our case, usually it starts from ~2 and slower decrease to 1.2~1.5 (roughly) I would suggest 
1. lowering the learning rate
2. when you do finetuning, try co-finetuning (mixing existing dataset w/ your own dataset)
3. unit test to make sure there is no bug in your code


To Jiapeng:
Training loss not decreasing doesn't mean the eval results also get stuck. I would suggest 1. having your own eval benchmark and 2. evaluating the latest checkpoint from time to time to see if there is progress on the training.

**Comment by Lyken17** [comment]
Hi there,

Our team has moved our VILA to a new location: [NVLabs/VILA](https://github.com/NVlabs/VILA). We kindly request that you update your references to the old repository location with the new one (e.g, git clone / pipeline integration). This will ensure that you have access to the latest updates, improvements, and bug fixes for VILA project. Current repo will be archieved in one week (Jul 10th) and made private after one month (Aug 4th).

Thank you for your understanding and continued support. If your issue still exists, feel free to open a new one in the new place. We appreciate your patience and cooperation during this transition.

Best regards,
VILA Team
